### PR TITLE
Create SCOPES.md

### PR DESCRIPTION
**Pull request by lisajguo**
Is this right? Let me know :)

### CODE REVIEW

File: SCOPES.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Prefixes
+Supported scope prefixes:
+* `project:releng:beetmover`
+  * Used for Firefox/Fennec/Devedition style beetmover tasks
+
+* `project:comm:thunderbird:releng:beetmover`
+  * Used for Thunderbird style balrog tasks
```
Comment (MihaiTabara) [inline SCOPES.md]: slippery keyboard: s/balrog/beetmover?

File: SCOPES.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+
+# Scopes
+* `{scope_prefix}:action:push-to-candidates`
+  * Tells beetmoverscript to Schedule a release to ship on balrog channel(s)
```
Comment (MihaiTabara) [inline SCOPES.md]: Not exactly. This is beetmover, so we're talking about artifacts transfer. I'd rephrase this to: "Tells beetmoverscript to copy to candidates directory in S3 the corresponding release artifacts" or something alike.

File: SCOPES.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  * **Conflicts**: with any other `{scope_prefix}:action:*`
+  * **Branch Restrictions**: `all-candidate-tasks`
```
Comment (MihaiTabara) [inline SCOPES.md]: I think there's an `s` missing there, given [this](https://hg.mozilla.org/mozilla-central/file/tip/taskcluster/taskgraph/util/scriptworker.py#l120) reference, hence "all-candidates-tasks"

File: SCOPES.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+* `{scope_prefix}:action:push-to-release`
```
Comment (MihaiTabara) [inline SCOPES.md]: I think this is missing an 's', it should be `push-to-releases` given [this](https://hg.mozilla.org/mozilla-central/file/tip/taskcluster/taskgraph/util/scriptworker.py#l120) reference.

File: SCOPES.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  * Tells beetmoverscript to submit locale information for a release.
```
Comment (MihaiTabara) [inline SCOPES.md]: Slippery copy-paste from balrogscript maybe? Beetmoverscript `push-to-releases` means we're copying artifacts from candidates directory to releases directory, hence I'd rephrase this to something like: "Tells beetmoverscript to move the artifacts from `candidates` directory to `releases` directory for the corresponding release"

File: SCOPES.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    * This is **default** if no `action` specified in scopes.
```
Comment (MihaiTabara) [inline SCOPES.md]: The default is actually the `push-to-nightly` provided by [this](https://hg.mozilla.org/mozilla-central/file/tip/taskcluster/taskgraph/util/scriptworker.py#l146) reference.

File: SCOPES.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  * **Conflicts**: with any other `{scope_prefix}:action:*`
+  * **Branch Restrictions**: `all-push-tasks`
+
+* `{scope_prefix}:action:push-to-nightly`
+  * Tells beetmoverscript to submit the initial toplevel release information.
```
Comment (MihaiTabara) [inline SCOPES.md]: Slippery copy-paste from balrogscript maybe? Again, this is beetmover we're talking  about so we're talking about artifacts transfer. I'd rephrase this to: "Tells beetmoverscript to copy to nightly directory in S3 the corresponding release artifacts" or something alike.

File: SCOPES.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  * **Conflicts**: with any other `{scope_prefix}:action:*`
+  * **Branch Restrictions**: `all-nightly-tasks`
+
```
Comment (MihaiTabara) [inline SCOPES.md]: There is also a `push-to-partner` missing here. It conflicts with any other action and "Tells beetmoverscript to move the corresponding artifacts of partner releases of Firefox to their corresonding private buckets"

File: SCOPES.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+* `{scope_prefix}:bucket:release`
+  * Tells beetmoverscript to use the aurora bucket config
```
Comment (MihaiTabara) [inline SCOPES.md]: "Tells beetmoverscript to use the production credentials to access the `release` buckets (namely `candidates` and `releases`)"

File: SCOPES.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    * *Informative*: This is production balrog, with aurora channel (devedition)
```
Comment (MihaiTabara) [inline SCOPES.md]: Slippery keyboard copy-paste from balrog? We can safely remove this line.

File: SCOPES.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  * **Conflicts**: with any other `{scope_prefix}:bucket:*`
+  * **Branch Restrictions**: `all-release-branches`
+
+* `{scope_prefix}:bucket:nightly`
+  * Tells beetmoverscript to use the nightly bucket config
```
Comment (MihaiTabara) [inline SCOPES.md]: I'd s/config/credentials or "Tells beetmoverscript to use the production credentials to access the `nightly` bucket"

File: SCOPES.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    * *Informative*: This is production balrog, with nightly channel
```
Comment (MihaiTabara) [inline SCOPES.md]: Slippery keyboard copy-paste from balrog? We can safely remove this line.

File: SCOPES.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+  * **Conflicts**: with any other `{scope_prefix}:bucket:*`
+  * **Branch Restrictions**: 
+    * `all-nightly-branches`
```
Comment (MihaiTabara) [inline SCOPES.md]: For consistency, let's move this on the same line as above.

File: SCOPES.md [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+* `{scope_prefix}:bucket:dep`
+  * Tells beetmoverscript to use the nightly bucket config
```
Comment (MihaiTabara) [inline SCOPES.md]: "Tells beetmoverscript to use the `dep` (which stands for dependent) credentials to access the staging buckets"

File: SCOPES.md [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+    * *Informative*: This is staging balrog, with any (otherwise valid) channel
```
Comment (MihaiTabara) [inline SCOPES.md]: Safe to remove this line, nothing to do with balrog.

File: SCOPES.md [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+  * **Conflicts**: with any other `{scope_prefix}:bucket:*`
+  * **Branch Restrictions**: `None`
+
```
Comment (MihaiTabara) [inline SCOPES.md]: There are also another two types of scopes missing here, from [puppet](https://hg.mozilla.org/build/puppet/file/tip/modules/beetmover_scriptworker/templates/prod_script_config.json.erb)

`partner` - "Tells beetmoverscript to use the production credentials to access the `partner` bucket"
`dep-partner` - same as above but for staging

File: SCOPES.md [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+* `project:comm:thunderbird:releng:beetmover`
+  * Used for Thunderbird style beetmover tasks
+
+
+# Scopes
+* `{scope_prefix}:action:push-to-candidates`
+  * Tells beetmoverscript to copy to candidates directory in S3 the corresponding release artifacts
+  * **Conflicts**: with any other `{scope_prefix}:action:*`
+  * **Branch Restrictions**: `all-candidates-tasks`
```
Comment (Callek) [inline SCOPES.md]: In this case there is no restriction, I've used **Branch Restrictions** as a means of saying what scriptworker will barf on, and less about what the taskgraph will produce.

And in doing so it gets the definitions from scriptworker, specifically https://github.com/mozilla-releng/scriptworker/blob/master/scriptworker/constants.py#L244 so if its missing in that set it's `None`, or it can be `all-nightly-branches` etc.

File: SCOPES.md [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+
+* `{scope_prefix}:action:push-to-release`
+  * Tells beetmoverscript to move the artifacts from candidates directory to releases directory for the corresponding release
+  * **Conflicts**: with any other `{scope_prefix}:action:*`
+  * **Branch Restrictions**: `all-push-tasks`
```
Comment (Callek) [inline SCOPES.md]: Restrictions here too (I'm going to omit commenting on other restriction lines in this review pass)

File: SCOPES.md [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+  * Tells beetmoverscript to move the corresponding artifacts of partner releases of Firefox to their corresonding private buckets
+  * **Conflicts**: with any other `{scope_prefix}:action:*`
+  * **Branch Restrictions**: `all-candidates-tasks`
+
+* `{scope_prefix}:bucket:release`
+  * Tells beetmoverscript to use the production credentials to access the release buckets (namely candidates and releases)
+  * **Conflicts**: with any other `{scope_prefix}:bucket:*`
+  * **Branch Restrictions**: `all-release-branches`
+
+* `{scope_prefix}:bucket:nightly`
```
Comment (Callek) [inline SCOPES.md]: I've gone with the loose desire to sort alpanumeric on the scopes (which would put `bucket:release` below `bucket:nightly` but I don't feel it is a big deal if we don't)

File: SCOPES.md (hunk 17 above)
Comment (lisajguo) [reply SCOPES.md]: Okay thank you, that makes sense, setting it to None
